modules/SS1/nlobjRecord.js

```diff
@@ -39,62 +43,39 @@ var nlobjRecord = function (recordtype, internalid) {
   }
 
   var getLineItemCount = function(group) {
-    if(group == 'item') {
-      return lineItems.length
-    } else if(group == 'addressbook') {
-      return addressBookLines.length
-    } else {
-      return 0
+    if(typeof sublists[group] !== 'undefined') {
+      return sublists[group].length;
     }
+    return 0;
   }
 
   var setLineItemValue = function(group,name,linenum,value) {
-    if(group == 'item') {
-      lineItems[linenum-1][name] = value
-    } else {
-      throw new Error('NETSIM ERROR: Line item group: '+group+' is unsupported.');
-    }
+    sublists[group][linenum-1][name] = value;
   }
 
   var getLineItemValue = function(group,name,line) {
-    if(group == 'item') {
-      return lineItems[line-1][name]
-    } else if(group == 'addressbook') {
-      return addressBookLines[line-1][name]
-    } else {
-      throw new Error('NETSIM ERROR: Line item group: '+group+' is unsupported.');
-    }
+    return sublists[group][line-1][name];
   }
 
   var findLineItemValue = function(group,name,value) {
-    if(group == 'item') {
-      for(var i = 0; i < lineItems.length; i++) {
-        if(lineItems[i][name] == value)
-          return i+1
-      }
-    } else if(group == 'addressbook') {
-      for(var i = 0; i < addressBookLines.length; i++) {
-        if(addressBookLines[i][name] == value)
-          return i+1
-      }
-    } else {
-      throw new Error('NETSIM ERROR: Line item group: '+group+' is unsupported.');
+    var sublist = sublists[group];
+    for(var i = 0; i < sublist.length; i++) {
+      if(sublist[i][name] == value)
+        return i+1
     }
     return -1
   }
 
   var selectNewLineItem = function(group) {
-    if(group == 'item') {
-      currentLineItems[group] = {}
-      currentLineItems[group]['id'] = id+'_'+lineItems.length;
-      currentLineItems[group]['line'] = lineItems.length;
-    } else if(group == 'addressbook') {
-      currentLineItems[group] = {}
-      currentLineItems[group]['id'] = id+'_'+getLineItemCount(group);
-      currentLineItems[group]['line'] = getLineItemCount(group);
+    if (typeof sublists[group] === 'undefined') {
+      sublists[group] = [];
+    }
+    currentLineItems[group] = {}
+    currentLineItems[group]['id'] = id+'_'+getLineItemCount(group);
+    currentLineItems[group]['line'] = getLineItemCount(group);
```

Comment:
Any particular reason you're constructing this object one property at a time, rather than just using an object literal?